src/main/java/de/photon/anticheataddition/util/config/ConfigurationRepresentation.java

```diff
@@ -49,44 +49,54 @@ public synchronized void requestValueChange(final String path, final Object valu
-    {
-        // Directly inject changes.
+    public synchronized void save() throws IOException {
         if (requestedChanges.isEmpty()) return;
 
-        // Load the whole config.
         final var configLines = new ArrayList<>(Files.readAllLines(this.configFile.toPath()));
 
         for (ConfigChange requestedChange : requestedChanges) {
-            final var value = requestedChange.value();
-
             final int lineIndexOfKey = requestedChange.lineIndexOfPath(configLines);
-            final String originalLine = configLines.get(lineIndexOfKey);
-            final int affectedLines = linesOfKey(configLines, lineIndexOfKey);
-
-            // We want to delete all lines after the initial one.
-            deleteLines(configLines, lineIndexOfKey + 1, affectedLines - 1);
-
-            // +1 to add the otherwise removed ':' right back.
-            final var replacementLine = new StringBuilder(originalLine.substring(0, originalLine.lastIndexOf(':') + 1));
-
-            // Set the new value.
-            if (value instanceof Boolean || value instanceof Number) replacementLine.append(' ').append(value);
-            else if (value instanceof String str) replacementLine.append(' ').append('\"').append(str).append('\"');
-            else if (value instanceof List<?> list) {
-                if (list.isEmpty()) replacementLine.append(" []");
-                else {
-                    final var preString = "- ".indent((int) ConfigUtil.depth(originalLine));
-                    for (Object o : list) configLines.add(lineIndexOfKey + 1, preString + o);
-                }
-            }
 
-            configLines.set(lineIndexOfKey, replacementLine.toString());
+            String replacementLine = createReplacementLine(configLines, requestedChange, lineIndexOfKey);
+            configLines.set(lineIndexOfKey, replacementLine);
+
+            if (requestedChange.value() instanceof List<?> list) {
+                handleListValue(configLines, list, lineIndexOfKey, requestedChange);
+            }
         }
 
         Files.write(this.configFile.toPath(), configLines);
     }
 
+    private String createReplacementLine(ArrayList<String> configLines, ConfigChange requestedChange, int lineIndexOfKey) {
+        final var value = requestedChange.value();
+        final String originalLine = configLines.get(lineIndexOfKey);
+        final int affectedLines = linesOfKey(configLines, lineIndexOfKey);
+
+        deleteLines(configLines, lineIndexOfKey + 1, affectedLines - 1);
+
+        final var replacementLine = new StringBuilder(originalLine.substring(0, originalLine.lastIndexOf(':') + 1));
+
+        if (value instanceof Boolean || value instanceof Number) {
+            replacementLine.append(' ').append(value);
+        } else if (value instanceof String str) {
+            replacementLine.append(' ').append('\"').append(str).append('\"');
+        } else if (value instanceof List<?>) {
+            replacementLine.append(" []");
```

Comment:
This will always add the empty list value to the config, and then afterwards add other lines. This is a bug.